zjit/src/hir.rs

```diff
@@ -644,10 +644,13 @@ pub enum Insn {
     IfTrue { val: InsnId, target: BranchEdge },
     IfFalse { val: InsnId, target: BranchEdge },
 
-    /// Call a C function
+    /// Call a C function without pushing a frame
     /// `name` is for printing purposes only
     CCall { cfun: *const u8, args: Vec<InsnId>, name: ID, return_type: Type, elidable: bool },
 
+    /// Call a C function that pushes a frame
+    CCallWithFrame {cfun: *const u8, args: Vec<InsnId>, cme: *const rb_callable_method_entry_t, name: ID, state: InsnId },
```

Comment:
It's not new in this HIR, but can we call it `cfunc` instead of `cfun`? It would be more consistent with the naming of `VM_METHOD_TYPE_CFUNC`.